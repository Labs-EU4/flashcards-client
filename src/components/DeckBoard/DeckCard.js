import React, {useState} from "react";
import {Card, Avatar, Icon} from "antd";
import {Link} from "react-router-dom";
import * as styles from "./deckCard.module.css";
import EditModal from "./EditModal";
const {Meta} = Card;

export default function DeckCard({deck, deleteDeck, updateDeck}) {
  const [visible, setVisible] = useState(false);
  const location = id => ({
    pathname: `/deck/${id}`,
    state: {source: "personal"},
  });
  return (
    <>
      <Card
        style={{width: 300}}
        actions={[
          <Link to={`/play/${deck.deck_id}`}>
            <Icon type="play-circle" data-testid="play" />
          </Link>,
          <Icon type="edit" onClick={() => setVisible(!visible)} data-testid="edit" />,
          <Icon
            type="delete"
            onClick={() => deleteDeck(deck.deck_id)}
            data-testid="delete"
          />,
        ]}
        className={styles.deckCard}
      >
        <Link to={location(deck.deck_id)}>
          <Meta avatar={<Avatar src="logo192.png" />} description={deck.deck_name} />
        </Link>
      </Card>
      <EditModal
        visible={visible}
        deckValues={deck}
        updateDeck={updateDeck}
        setVisible={setVisible}
      />
    </>
  );
}
