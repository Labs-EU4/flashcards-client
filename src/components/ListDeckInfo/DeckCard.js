import React, {useState} from "react";
import {Icon, Popover, Button} from "antd";
import styles from "./DeckList.module.css";

export default function DeckCard({deck}) {
  const [visible, setVisible] = useState(false);

  function hide() {
    setVisible(false);
  }

  function handleVisibleChange() {
    setVisible(!visible);
  }

  return (
    <div className={styles.deckCard}>
      <div className={styles.deckOverview}>
        <div className={styles.header}>
          <h2>{deck.deck_name}</h2>
          <div className={styles.info}>
            <h3 className={styles.info_headings}>Author: {deck.author}</h3>
            <h3 className={styles.info_headings}>
              No. of Cards: {deck.flashcards.length}
            </h3>
          </div>
        </div>
        <div className={styles.actions}>
          <Icon type="like" style={{color: "rgba(0,0,0,.25)"}} />
          <Icon type="dislike" style={{color: "rgba(0,0,0,.25)"}} />
          <Popover
            placement="bottom"
            content={
              <>
                <div className={styles.gameOptions}>
                  <h4>Play through</h4>
                  <Button>Input</Button>
                  <Button>Cards</Button>
                </div>
                <Button type="primary" onClick={hide}>
                  Start
                </Button>
              </>
            }
            title="Title"
            trigger="click"
            visible={visible}
            onVisibleChange={handleVisibleChange}
          >
            <Icon type="more" style={{color: "rgba(0,0,0,.25)"}} />
          </Popover>

          <Icon type="star" style={{color: "rgba(0,0,0,.25)"}} />
        </div>
      </div>
      {/* <div className={styles.horizontalList}>
        {deck.flashcards.map(card => {
          return (
            <Card title="Question" className={styles.card}>
              {card.question}
            </Card>
          );
        })}
      </div> */}
    </div>
  );
}
