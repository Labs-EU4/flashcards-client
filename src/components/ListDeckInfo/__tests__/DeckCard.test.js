import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DeckCard from "../DeckCard";

let mockDeck = {
  deck_name: "Test Deck",
  author: "John Doe",
  flashcards: ["just", "mock", "data"],
};

const DeckName = () => {
  return wrapper.queryByTestId("deck_name");
};

const Author = () => {
  return wrapper.queryByTestId("author");
};

const FlashcardsCount = () => {
  return wrapper.queryByTestId("flashcards_count");
};

const MoreOptions = () => {
  return wrapper.queryByTestId("more_options");
};

const Popover = () => {
  return wrapper.queryByTestId("popover");
};

const LikeButton = () => {
  return wrapper.queryByTestId("like");
};

const DislikeButton = () => {
  return wrapper.queryByTestId("dislike");
};

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(<DeckCard deck={mockDeck} />);
});

it("renders without crashing", () => {
  expect(wrapper.container).toMatchSnapshot();
});

it("displays elements correctly", () => {
  expect(DeckName()).toBeInTheDocument();
  expect(DeckName()).toHaveTextContent("Test Deck");
  expect(Author()).toBeInTheDocument();
  expect(Author()).toHaveTextContent("Author: John Doe");
  expect(FlashcardsCount()).toBeInTheDocument();
  expect(FlashcardsCount()).toHaveTextContent("No. of Cards: 3");
  expect(MoreOptions()).toBeInTheDocument();
  expect(LikeButton()).toBeInTheDocument();
  expect(DislikeButton()).toBeInTheDocument();
  expect(Popover()).not.toBeInTheDocument();
});

it("displays popover when more button is clicked", () => {
  rtl.fireEvent.click(MoreOptions());
  expect(Popover()).toBeInTheDocument();
  expect(Popover()).toBeVisible();
});
