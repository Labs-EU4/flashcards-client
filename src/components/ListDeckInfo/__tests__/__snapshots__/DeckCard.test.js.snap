// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="deckCard"
    data-testid="deck_card_container"
  >
    <div
      class="deckOverview"
    >
      <div
        class="header"
      >
        <h2
          data-testid="deck_name"
        >
          Test Deck
        </h2>
        <div
          class="info"
        >
          <h3
            class="info_headings"
            data-testid="author"
          >
            Author: 
            John Doe
          </h3>
          <h3
            class="info_headings"
            data-testid="flashcards_count"
          >
            No. of Cards: 
            3
          </h3>
        </div>
      </div>
      <div
        class="actions"
      >
        <i
          aria-label="icon: like"
          class="anticon anticon-like"
          data-testid="like"
          style="color: rgba(0, 0, 0, 0.25);"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="like"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 0 0-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 0 0 471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 0 1 6.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0 1 42.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: dislike"
          class="anticon anticon-dislike"
          data-testid="dislike"
          style="color: rgba(0, 0, 0, 0.25);"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="dislike"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M885.9 490.3c3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-51.6-30.7-98.1-78.3-118.4a66.1 66.1 0 0 0-26.5-5.4H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h129.3l85.8 310.8C372.9 889 418.9 924 470.9 924c29.7 0 57.4-11.8 77.9-33.4 20.5-21.5 31-49.7 29.5-79.4l-6-122.9h239.9c12.1 0 23.9-3.2 34.3-9.3 40.4-23.5 65.5-66.1 65.5-111 0-28.3-9.3-55.5-26.1-77.7zM184 456V172h81v284h-81zm627.2 160.4H496.8l9.6 198.4c.6 11.9-4.7 23.1-14.6 30.5-6.1 4.5-13.6 6.8-21.1 6.7a44.28 44.28 0 0 1-42.2-32.3L329 459.2V172h415.4a56.85 56.85 0 0 1 33.6 51.8c0 9.7-2.3 18.9-6.9 27.3l-13.9 25.4 21.9 19a56.76 56.76 0 0 1 19.6 43c0 9.7-2.3 18.9-6.9 27.3l-13.9 25.4 21.9 19a56.76 56.76 0 0 1 19.6 43c0 9.7-2.3 18.9-6.9 27.3l-14 25.5 21.9 19a56.76 56.76 0 0 1 19.6 43c0 19.1-11 37.5-28.8 48.4z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: more"
          class="anticon anticon-more"
          data-testid="more_options"
          style="color: rgba(0, 0, 0, 0.25);"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="more"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M456 231a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 280a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 280a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: star"
          class="anticon anticon-star"
          style="color: rgba(0, 0, 0, 0.25);"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="star"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
            />
          </svg>
        </i>
      </div>
    </div>
  </div>
</div>
`;
